### DISCUSSION TITLE
Proper parameter to import ECDSA(JWK) private key

### DISCUSSION THREAD

**Discussion by jasonclg** [panva/jose#416] [Q&A] [answered]
I was able to import JWK private key based on the [ecPublicKey example](https://github.com/panva/jose/blob/main/docs/functions/key_import.importJWK.md#readme)
```
  const privateKey = await importJWK(
    {
      crv: "P-256",
      kty: "EC",
      x: "MKBCTNIcKUSDii11ySs3526iDZ8AiTo7Tu6KPAqv7D4",
      y: "4Etl6SRW2YiLUrN5vfvVHuhp7x8PxltmWWlbbM4IFyM",
      d: "870MB6gfuTJ4HtUnUvYMyJpr5eUZNP4Bk43bVdj3eAE",
    },
    "ES256"
  );
```
However, if I remove x and y (coordinates of public key), it will throw `DOMException: The required JWK member "x" was missing
`
My question: why is public key required to import JWK private key?

**Comment by panva** [comment, marked as answer]
Because that's how the different specifications are defined and layered. 

The JWK definition requires public key components to always be present. Web Crypto API checks the input to be a valid JWK.
